Warning Wdelete-non-abstract-non-virtual-dtor

Hello,
I am using ginkgo/1.4.0 through Conan, and when, in c++, on Mac, I include `#include <ginkgo/ginkgo.hpp>`, then I get a warning:

```
/opt/homebrew/opt/llvm/bin/../include/c++/v1/__memory/unique_ptr.h:57:5: error: delete called on non-final 'gko::log::Record' that has virtual functions but non-virtual destructor [-Werror,-Wdelete-non-abstract-non-virtual-dtor]
    delete __ptr;
```

I tried the following but got the same error:
```
#ifdef _WIN32
#pragma warning(push, 0)
#else
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Weverything"
#pragma GCC diagnostic ignored "-Wdelete-non-abstract-non-virtual-dtor"
#endif
#include <ginkgo/ginkgo.hpp>
#ifdef _WIN32
#pragma warning(pop)
#else
#pragma GCC diagnostic pop
#endif
```
Any suggestion?
Note that I would like to continue to treat warnings as errors in my project.
Thank you.
Best regards!

--- a/include/ginkgo/core/base/executor.hpp
+++ b/include/ginkgo/core/base/executor.hpp
@@ -1480,8 +1480,8 @@ protected:
                  bool device_reset = false,
                  allocation_mode alloc_mode = default_cuda_alloc_mode)
         : EnableDeviceReset{device_reset},
-          alloc_mode_{alloc_mode},
-          master_(master)
+          master_(master),
+          alloc_mode_{alloc_mode}
     {
         this->get_exec_info().device_id = device_id;
         this->get_exec_info().num_computing_units = 0;
@@ -1685,8 +1685,8 @@ protected:
                 bool device_reset = false,
                 allocation_mode alloc_mode = default_hip_alloc_mode)
         : EnableDeviceReset{device_reset},
-          alloc_mode_(alloc_mode),
-          master_(master)
+          master_(master),
+          alloc_mode_(alloc_mode)
     {
         this->get_exec_info().device_id = device_id;
         this->get_exec_info().num_computing_units = 0;

--- a/include/ginkgo/core/log/logger.hpp
+++ b/include/ginkgo/core/log/logger.hpp
@@ -531,6 +531,8 @@ public:
     static constexpr mask_type criterion_events_mask =
         criterion_check_started_mask | criterion_check_completed_mask;
 
+    virtual ~Logger() = default;
+
 protected:
     /**
      * Constructor for a Logger object.

--- a/include/ginkgo/core/matrix/csr.hpp
+++ b/include/ginkgo/core/matrix/csr.hpp
@@ -186,6 +186,8 @@ public:
          */
         strategy_type(std::string name) : name_(name) {}
 
+        virtual ~strategy_type() = default;
+
         /**
          * Returns the name of strategy
          *

--- a/include/ginkgo/core/stop/residual_norm.hpp
+++ b/include/ginkgo/core/stop/residual_norm.hpp
@@ -98,8 +98,8 @@ protected:
                               remove_complex<ValueType> reduction_factor,
                               mode baseline)
         : EnablePolymorphicObject<ResidualNormBase, Criterion>(exec),
-          device_storage_{exec, 2},
           reduction_factor_{reduction_factor},
+          device_storage_{exec, 2},
           baseline_{baseline},
           system_matrix_{args.system_matrix},
           b_{args.b},

--- a/test/test_install/test_install.cpp
+++ b/test/test_install/test_install.cpp
@@ -328,7 +328,7 @@ int main()
 
     // core/log/convergence.hpp
     {
-        auto test = gko::log::Convergence<>::create(exec);
+        auto test = gko::log::Convergence<>::create();
     }
 
     // core/log/record.hpp
@@ -338,13 +338,13 @@ int main()
 
     // core/log/stream.hpp
     {
-        auto test = gko::log::Stream<>::create(exec);
+        auto test = gko::log::Stream<>::create();
     }
 
 #if GKO_HAVE_PAPI_SDE
     // core/log/papi.hpp
     {
-        auto test = gko::log::Papi<>::create(exec);
+        auto test = gko::log::Papi<>::create();
     }
 #endif  // GKO_HAVE_PAPI_SDE
 

